Map tool idenfify: Add support for index in mesh data

ptitjano opened the pull request:
## Description

This extends the identify tool to also display the index of a vertex/face/edge of a mesh.
See, the screenshot below: 

![image](https://github.com/user-attachments/assets/12dcbc35-ab7b-434f-8011-dfee214fa074)



Funded by: EDF

ptitjano committed:
qgsmeshlayer: Introduce closestElement

This is a generalization of `snapOnElement`, since it allows to also
get the element (face, vertex or edge) index.

ptitjano committed:
qgsmaptoolidentify: Display index for mesh data

This is achieved by using `QgsMeshLayer::closestElement()` instead of
`QgsMeshLayer::snapOnElement()`.

nyalldawson commented on src/core/mesh/qgsmeshlayer.h:
```diff
@@ -720,6 +720,35 @@ class CORE_EXPORT QgsMeshLayer : public QgsMapLayer, public QgsAbstractProfileSo
+      *
+      * \since QGIS 3.44
+      */
+    int closestElement( QgsMesh::ElementType elementType, const QgsPointXY &point, double searchRadius, QgsPointXY &projectedPoint ) const;
```
```suggestion
    int closestElement( QgsMesh::ElementType elementType, const QgsPointXY &point, double searchRadius, QgsPointXY &projectedPoint SIP_OUT ) const;
```

nyalldawson commented on src/core/mesh/qgsmeshlayer.cpp:
```diff
@@ -925,38 +926,16 @@ int QgsMeshLayer::closestEdge( const QgsPointXY &point, double searchRadius, Qgs
   return selectedIndex;
 }
 
-QgsMeshDatasetIndex QgsMeshLayer::staticVectorDatasetIndex( int group ) const
```
Why move this code? It clutters the diff here, and will make git blame a bit more frustrating...

ptitjano replied:
This MR introduces `closestFace`, `closestVertex` (which replace `snapOnFace` and `snapOnVertex` on top of an existing `closestEdge` method.  The existing `closestEdge` is in the middle of other functions. The idea if to regroup  `closestFace`, `closestVertex` and `closestEdge` together. That's why,  `closestEdge` is moved and this clutters the diff.

To solve it, I have moved the existing `closestEdge` first in a separate commit. This way, the diff of the `closestElement` commit is cleaner.

ptitjano committed:
qgsmeshlayer: Move closestEdge function closer to snap functions

This commit does not do any functional change. It moves the existing
`closestEdge()` method closer to `snapOnVertex` and
`snapOnFace`. Indeed, the following commits will introduce functional
changes to these functions by introducing `closestFace` and
`closestVertex`.

ptitjano committed:
qgsmeshlayer: Ensure to have an empty closesPoint if search fails

With this change, `projectedPoint` is reset to an empty QgsPointXY
before the search begins. This ensures that an empty point is returned
is the search fails.
